source/extensions/filters/common/expr/context.h

```diff
@@ -108,7 +111,14 @@ class BaseWrapper : public google::api::expr::runtime::CelMap,
   const google::api::expr::runtime::CelList* ListKeys() const override {
     NOT_IMPLEMENTED_GCOVR_EXCL_LINE;
   }
-  CelValue Produce(ProtobufWkt::Arena*) override { return CelValue::CreateMap(this); }
+  CelValue Produce(ProtobufWkt::Arena* arena) override {
+    // Producer is unique per evaluation arena since activation is re-created.
+    arena_ = arena;
+    return CelValue::CreateMap(this);
+  }
+
+protected:
+  ProtobufWkt::Arena* arena_;
```

Comment:
How is this being used now?

Reply:
`serializeString` returns a string by value, so a copy is made on this arena, because CelValue takes `const string*`.

Reply:
Maybe I'm missing something, but this looks like just a pointer to the arena that gets assigned above, how is the copy done?

Reply:
`arena_` is used in the implementation of the wrapper. The base class is shared with other wrappers, which is how it gets passed down. I am working on making this flow better upstream, and get rid of the arena use in this case, but it's not simple.